sda-commons-web-testing/src/main/java/org/sdase/commons/spring/boot/web/testing/mongodb/ExternalMongoAutoConfiguration.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2022- SDA SE Open Industry Solutions (https://www.sda.se)
+ *
+ * Use of this source code is governed by an MIT-style
+ * license that can be found in the LICENSE file or at
+ * https://opensource.org/licenses/MIT.
+ */
+package org.sdase.commons.spring.boot.web.testing.mongodb;
+
+import java.util.Map;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.env.EnvironmentPostProcessor;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.MapPropertySource;
+
+public class ExternalMongoAutoConfiguration implements EnvironmentPostProcessor {
+  private static final String FLAPDOODLE_AUTO_CONFIG_CLASS =
+      "de.flapdoodle.embed.mongo.spring.autoconfigure.EmbeddedMongoAutoConfiguration";
+  private static final String TEST_MONGODB_CONNECTION_STRING_PROPERTY_NAME =
+      "test.mongodb.connection.string";
+  private static final String AUTOCONFIGURE_EXCLUDES_PROPERTY_NAME = "spring.autoconfigure.exclude";
+  private static final String SPRING_DATA_MONGODB_URI_PROPERTY_NAME = "spring.data.mongodb.uri";
```

Comment:
Can we make the same approach without introducing our own property? Can't we implement the same behaviour if we pass `SPRING_DATA_MONGODB_URI` from outside?

Reply:
It is by intention to use the same environment variable as [in Dropwizard Commons](https://github.com/SDA-SE/sda-dropwizard-commons/blob/master/sda-commons-server-mongo-testing/src/main/java/org/sdase/commons/server/mongo/testing/MongoDb.java#L43). That's also the default in our build pipelines. We would require the same code because Flapdoodle's auto configuration does not care about any settings you apply.

Reply:
And the environment variable is also named in the task 😉

Reply:
Ok. The Jenkins Pipeline will work out-of-the-box in this case. Let's keep it!